plugins/inputs/win_perf_counters/pdh.go

```diff
@@ -414,11 +469,64 @@ func PdhValidatePath(path string) uint32 {
 	return uint32(ret)
 }
 
+func errnoErr(e syscall.Errno) error {
+	switch e {
+	case 0:
+		return nil
+	case errnoERROR_IO_PENDING:
+		return errERROR_IO_PENDING
+	}
+	// TODO: add more here, after collecting data on the common
+	// error values see on Windows. (perhaps when running
+	// all.bat?)
+	return e
+}
+
+// The windows native call for converting a 16-bit wide character string (UTF-16) to a null terminated string.
+//
+// Note: If you call the function and not pass in an out string, the return value will be the length of the
+// input string.
+// Example usage:
+//   cc, err := WideCharToMultiByte(65001, 0, s, -1, nil, 0)
+//   if err != nil {
+// 	  fmt.Println("CONVERSION ERROR: ", err)
+//   }
+//
+//	 fmt.Println("Length bytes: ", cc)
+//   n, err := WideCharToMultiByte(65001, 0, s, 1<<29, &outStr[0], 1<<29)
+//   if err != nil {
+// 	   fmt.Println("CONVERSION ERROR: ", err)
+//   }
+//   fmt.Println("Converted bytes: ", n)
+//
+func WideCharToMultiByte(codePage uint32, dwFlags uint32, wchar *uint16, nwchar int32, str *byte, nstr int32) (nwrite int32, err error) {
```

Comment:
Where is this called from?

Reply:
It's not. It was one of those things I used to try and fix the win32 bug without padding the structures. The call itself works good and I thought it might be useful in the future. I've noted this in a comment, but I can remove it as well.